vm/actor/src/builtin/market/mod.rs

```diff
@@ -265,7 +265,10 @@ impl Actor {
         let mut state: State = rt.state::<State>()?;
 
         let store = rt.store();
-        let msm = state.mutator(store);
+        let mut msm = state.mutator(store);
+        msm.with_pending_proposals(Permission::ReadOnly);
+        msm.with_escrow_table(Permission::ReadOnly);
+        msm.with_locked_table(Permission::ReadOnly);
```

Comment:
Not tested but pretty sure you can chain withs like this:

```
let msm = state.mutator(store)
    .with_pending_proposals(Permission::ReadOnly)
    .with_escrow_table(Permission::ReadOnly)
    .with_locked_table(Permission::ReadOnly);
```

This would allow `msm` binding staying immutable.

Reply:
hm, i get an error when i try that:
```
error[E0716]: temporary value dropped while borrowed
   --> vm/actor/src/builtin/market/mod.rs:268:23
    |
268 |           let msm = state.mutator(store)
    |                         ^^^^^^^^^^^^^^^^^^^^ creates a temporary which is freed while still in use
...
271 |           .with_locked_table(Permission::ReadOnly);
    |                                                   - temporary value is freed at the end of this statement
...
319 |               let client_balance_ok = msm
    |  _____________________________________-
320 | |                 .balance_covered(client, total_client_lockup.get(&client).unwrap())
    | |___________________________________________________________________________________- borrow later used here
    |
    = note: consider using a `let` binding to create a longer lived value
```

Reply:
There are places like [here](https://github.com/ChainSafe/forest/blob/c53d8794477625f4fbb9189b2b2275b9bf2f8a5c/vm/actor/src/builtin/market/mod.rs#L604) where withs are chained but then a `build` is used. Not sure at all if we need this.

Reply:
oh nice, yeah updated!